### PR TITLE
BASWSPRT-166: Fix for Frozen Screen in Activity Feed

### PR DESCRIPTION
**Pull request by deb1990**
## Overview
When an Activity is created with many(example 1000) Target Contacts, the Activity Feed gets frozen. This happens with Activities like "Mass SMS" etc.

Now, only first 25 records are shown in the UI, so that this performance issue does not happen anymore.

Also previously, for exactly similar issues, 'Bulk Email' activities were hidden, but now that the fix is global, those changes reverted from.
https://github.com/compucorp/uk.co.compucorp.civicase/pull/264
https://github.com/compucorp/uk.co.compucorp.civicase/pull/265
https://github.com/compucorp/uk.co.compucorp.civicase/pull/267

## How it looks
2021-01-21 at 5 26 PM
(in this case, the limit before hiding contacts is set to 3 to demonstrate the feature)

## Technical Details
1. A new API, `Activity.getAll` has been created. This API is similar to “Activity.get”. But if the 'target_contact_id' are more than 25, it returns first 25 only. Also adds a parameter `total_target_contacts`,  which contains the total number of contacts before hiding.

2. All instances where `Activity.get` was used, is replaced by `Activity.getAll`.

3. `Contact-Card` directive now accepts a `totalContacts` attribute. If present, it shows a notification `Click the Activity for all Contacts.`

4. All changed are reverted from
https://github.com/compucorp/uk.co.compucorp.civicase/pull/264
https://github.com/compucorp/uk.co.compucorp.civicase/pull/265
https://github.com/compucorp/uk.co.compucorp.civicase/pull/267

### CODE REVIEW

File: ang/civicase/contact/directives/contact-card.directive.html [hunk 1]
```diff
@@ -80,6 +80,11 @@
           {{contact[moreField]}}
         </div>
       </li>
+      <li ng-if="totalContacts">
+        <span class="civicase__contact-additional__hidden_contacts_info">
+          Click the Activity for all Contacts.
```
Comment (reneolivo) [inline ang/civicase/contact/directives/contact-card.directive.html:85]: use `ts`

Comment (reneolivo) [reply ang/civicase/contact/directives/contact-card.directive.html:85]: also, "for all Contacts", is it "to view all contacts" ?

**Review by reneolivo — CHANGES_REQUESTED** [review]

**Review by tunbola — APPROVED** [review]

**Review by reneolivo — APPROVED** [review]

File: api/v3/Activity/Getall.php [hunk 2]
```diff
@@ -33,18 +33,36 @@ function civicrm_api3_activity_getall(array $params) {
...
+
+/**
+ * Limit contacts.
+ *
+ * @param array $result
+ *   Results from which fields should be limited.
+ * @param string $fieldName
+ *   Name of the field to limit.
+ */
+function _limitContacts(array &$result, string $fieldName) {
```
Comment (reneolivo) [inline api/v3/Activity/Getall.php:51]: Please rename to `civicrm_api3_activity_getall_limitContacts`

File: ang/test/civicase/contact/directives/contact-card.directive.spec.js [hunk 3]
```diff
@@ -69,6 +69,20 @@
         });
       });
 
+      describe('when the display name does not have first name or last name but an email id either', function () {
```
Comment (reneolivo) [inline ang/test/civicase/contact/directives/contact-card.directive.spec.js:72]: I don't know if this statement is correct. Is this related to the `display_name` being empty? It seems you wanted to describe something similar to `when the contact doesn't have a a first or last name, but does have a display name`.

File: ang/test/civicase/contact/directives/contact-card.directive.spec.js [hunk 4]
```diff
@@ -69,6 +69,20 @@
...
+          spyOn(ContactsCache, 'getCachedContact').and.returnValue({
+            display_name: 'John Doe'
+          });
+          compileDirective(true, ContactsData.values[0].id);
+        });
+
+        it('sets the initial of the name as the avatar', function () {
+          expect(element.isolateScope().contacts[0].avatar).toBe('JD');
+        });
+      });
```
Comment (reneolivo) [inline ang/test/civicase/contact/directives/contact-card.directive.spec.js:84]: Looking at the logic implemented in `getInitials`, aren't we missing a test for contacts with a single display name?

**Review by reneolivo — CHANGES_REQUESTED** [review]

File: ang/test/civicase/contact/directives/contact-card.directive.spec.js (hunk 3 above)
Comment (deb1990) [reply ang/test/civicase/contact/directives/contact-card.directive.spec.js]: Fixed

File: ang/test/civicase/contact/directives/contact-card.directive.spec.js (hunk 4 above)
Comment (deb1990) [reply ang/test/civicase/contact/directives/contact-card.directive.spec.js:84]: As discussed this is not needed.

**Review by reneolivo — APPROVED** [review]

**Review by lisandro-compucorp — APPROVED** [review]
👍
